### PR TITLE
deps: upgrade llamaindex to 0.10.28

### PR DESCRIPTION
**Pull request by nicoloboschi**

### CODE REVIEW

File: tests/unit-tests/test_ragstack.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 def test_import():
-    from llama_index.vector_stores import AstraDBVectorStore  # noqa
-    from llama_index.vector_stores import CassandraVectorStore  # noqa
+    from llama_index.vector_stores.astra_db import AstraDBVectorStore  # noqa
```
Comment (cbornet) [inline tests/unit-tests/test_ragstack.py:2]: They removed it from the `__init__` ? That's a breaking change...

Comment (nicoloboschi) [reply tests/unit-tests/test_ragstack.py:2]: YES ! 
that's why I want to cut ragstack 1.0.0...

**Review by cbornet — APPROVED** [review]

**Comment by nicoloboschi** [comment]
waiting for https://github.com/run-llama/llama_index/pull/11896 to be released

**Comment by cbornet** [comment]
There's a conflict in transitive dependencies between langchain and llama-index...
```
Because langchain-google-genai (0.0.11) @ git+https://github.com/langchain-ai/langchain-google.git@main#subdirectory=libs/genai depends on google-generativeai (^0.4.1)
 and llama-index-embeddings-gemini (0.1.4) depends on google-generativeai (>=0.3.2,<0.4.0), langchain-google-genai (0.0.11) @ git+https://github.com/langchain-ai/langchain-google.git@main#subdirectory=libs/genai is incompatible with llama-index-embeddings-gemini (0.1.4).
```

**Comment by nicoloboschi** [comment]
> There's a conflict in transitive dependencies between langchain and llama-index...
> 
> ```
> Because langchain-google-genai (0.0.11) @ git+https://github.com/langchain-ai/langchain-google.git@main#subdirectory=libs/genai depends on google-generativeai (^0.4.1)
>  and llama-index-embeddings-gemini (0.1.4) depends on google-generativeai (>=0.3.2,<0.4.0), langchain-google-genai (0.0.11) @ git+https://github.com/langchain-ai/langchain-google.git@main#subdirectory=libs/genai is incompatible with llama-index-embeddings-gemini (0.1.4).
> ```

https://github.com/run-llama/llama_index/pull/12085

note this is only for langchain dev, but yeah, we need that before merging this PR

**Comment by nicoloboschi** [comment]
Update to 0.10.22 now, let's see if everything passes

File: ragstack-e2e-tests/pyproject.llamaindex.toml [hunk 2]
```diff
@@ -24,7 +24,7 @@ nemoguardrails = "^0.8.0"
 beautifulsoup4 = "^4"
 
 llama-index = { git = "https://github.com/run-llama/llama_index.git", branch = "main" }
-llama-index-readers-llama-parse = { git = "https://github.com/run-llama/llama_index.git", branch = "main", subdirectory = "llama-index-integrations/readers/llama-index-readers-llama-parse" }
+llama-index-readers-llama-parse = { git = "https://github.com/nicoloboschi/llama_index.git", branch = "llama-parse", subdirectory = "llama-index-integrations/readers/llama-index-readers-llama-parse" }
```
Comment (cbornet) [inline ragstack-e2e-tests/pyproject.llamaindex.toml:27]: See https://github.com/run-llama/llama_index/pull/12232

Comment (nicoloboschi) [reply ragstack-e2e-tests/pyproject.llamaindex.toml:27]: ah, I did the same PR then https://github.com/run-llama/llama_index/pull/12236

**Comment by nicoloboschi** [comment]
so they reverted llama-index gemini upgrade with https://github.com/run-llama/llama_index/pull/12126 ... 

let's wait for more clarity

**Pull request merged**